bed/enrichment.go

```diff
@@ -0,0 +1,232 @@
+package bed
+
+import (
+	"github.com/vertgenlab/gonomics/numbers"
+	"log"
+	"math"
+	"strings"
+)
+
+//subtractFromBedCoord is a helper function of overlapProbability that subtracts the values subStart and subEnd from entries in a slice of Bed b.
+//bClone is a clone of b used to save on memory allocation.
+func subtractFromBedCoord(b []Bed, subStart int, subEnd int, bClone []Bed) {
+	var prevEnd int = 0
+	var prevChrom string = ""
+
+	for i := range b {
+		if strings.Compare(prevChrom, "") == 0 || strings.Compare(prevChrom, bClone[i].Chrom) != 0 {
```

Comment:
We should really not be using strings.Compare. This should be 
`preChrom == "" || prevChrom != bClone[i].Chrom`

Reply:
agree